### PR TITLE
Add a method to preview post/page at editor

### LINKED ISSUE #915: Post/Page Preview
Simply open a big modal and include an iframe to let the user see the preview of the page in 3 modes:

* Desktop
* Tablet
* Phone

### PR DESCRIPTION
**Pull request by ArsalanSavand**
Fixes #915 

- Preview modal final
image

### CODE REVIEW

**Comment by ArsalanSavand** [comment]
@Alir3z4 everything works perfectly, but I get this error at console:
``Invalid 'X-Frame-Options' header encountered when loading 'http://skumfuk.draft.gonevis.com/when-im-small-by-phantogram/': 'ALLOW-FROM http://dash.draft.gonevis.com/#!/' is not a recognized directive. The header will be ignored.``

File: src/style/css/main.css [hunk 1]
```diff
@@ -1929,3 +1929,22 @@ img.cache {
     left: 250px !important;
     width: calc(100% - 250px) !important;
 }
+
+
+/* iframe */
+
+iframe {
```
Comment (AmirSavand) [inline src/style/css/main.css]: Don't add global selector.

Comment (Alir3z4) [reply src/style/css/main.css]: @AmirSavand care to explain why ?

Comment (AmirSavand) [reply src/style/css/main.css]: This style is meant for the post preview iframe not for all iframe on dashboard.

Add a class to it and change the selector.

**Comment by Alir3z4** [comment]
> @Alir3z4 everything works perfectly, but I get this error at console: ...

@ArsalanSavand 
Was this error happening before making ALLOW-FROM by backend ? or iframe wasn't possible at all ?
Also where is this happening ? is your frontend running on dashboard and API being on staging ?

Also, please make a bug report on the backend so I can start looking into it better.

File: src/app/entry/preview_modal/preview_modal_controller.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+import app from "../../app";
+
+function PreviewModalController($scope, URL, $sce) {
```
Comment (Alir3z4) [inline src/app/entry/preview_modal/preview_modal_controller.js]: This is missing documentation.
Please add some documentation what it does and where is being used.

Comment (Alir3z4) [reply src/app/entry/preview_modal/preview_modal_controller.js]: The document should also cover the `URL` parameter and its usage as well

Comment (Alir3z4) [inline src/app/entry/preview_modal/preview_modal_controller.js]: This is missing documentation.
Please add some documentation what it does and where is being used.

The document should also cover the `URL` parameter and its usage as well

File: src/app/entry/entry_edit/entry_edit_view.html [hunk 3]
```diff
@@ -18,8 +18,8 @@
         </button>
 
         <!-- Preview -->
-        <a class="btn btn-default" title="{{ 'VIEW_ON_SITE' | translate }}" target="_blank"
-          ng-href="{{ form.url }}" ng-if="editing">
+        <a ng-if="postInitial && editing" class="btn btn-default" title="{{ 'VIEW_ON_SITE' | translate }}"
```
Comment (Alir3z4) [inline src/app/entry/entry_edit/entry_edit_view.html]: This is no longer "View on Site"
This should be called "Preview" now.

**Comment by Alir3z4** [comment]
Great job on this Arsalan.

I just left some comments for you to take care.
On the preview modal itself, add another button _(blue color)_ saying "View on Site" which takes user to the document on the website in the preview mode.

There are other things to consider.

**Comment by ArsalanSavand** [comment]
@Alir3z4 alright...

File: src/style/css/main.css (hunk 1 above)
Comment (ArsalanSavand) [reply src/style/css/main.css]: Alright.

File: src/app/entry/entry_edit/entry_edit_view.html (hunk 3 above)
Comment (ArsalanSavand) [reply src/app/entry/entry_edit/entry_edit_view.html]: Done.

File: src/app/entry/preview_modal/preview_modal_controller.js (hunk 2 above)
Comment (ArsalanSavand) [reply src/app/entry/preview_modal/preview_modal_controller.js]: Done.

File: src/style/css/main.css (hunk 1 above)
Comment (ArsalanSavand) [reply src/style/css/main.css]: Done.

**Comment by ArsalanSavand** [comment]
Good to merge...

**Comment by Alir3z4** [comment]
Please don't merge this until you've answered https://github.com/SavandBros/gonevis-dash/pull/950#issuecomment-406805990

File: src/app/entry/entry_edit/entry_edit_view.html [hunk 4]
```diff
@@ -18,8 +18,8 @@
         </button>
 
         <!-- Preview -->
-        <a class="btn btn-default" title="{{ 'VIEW_ON_SITE' | translate }}" target="_blank"
-          ng-href="{{ form.url }}" ng-if="editing">
+        <a ng-if="postInitial && editing" class="btn btn-default" title="{{ 'PREVIEW' | translate }}"
```
Comment (Alir3z4) [inline src/app/entry/entry_edit/entry_edit_view.html]: Please confirm `PREVIEW` is in the translation file.

Comment (ArsalanSavand) [reply src/app/entry/entry_edit/entry_edit_view.html]: It was already added.

**Comment by AmirSavand** [comment]
@Alir3z4 approve this when you are ready.

**Comment by Alir3z4** [comment]
> @Alir3z4 approve this when you are ready.

@AmirSavand I'm more concerned about this https://github.com/SavandBros/gonevis-dash/pull/950#issuecomment-407177765 

Any update ?

**Comment by AmirSavand** [comment]
@ArsalanSavand review this comment https://github.com/SavandBros/gonevis-dash/pull/950#issuecomment-406805990

**Comment by ArsalanSavand** [comment]
@Alir3z4 about the error, this error occur after you fixed it.

**Comment by ArsalanSavand** [comment]
Before you attended to fix it, it was another error which we fixed it. But this is new.

**Comment by AmirSavand** [comment]
@Alir3z4 check comment and merge.

**Comment by Alir3z4** [comment]
I'll fix this error from backend and will merge this myself.

Are you blocked without merging this?

**Comment by AmirSavand** [comment]
No but you need to approve before I merge.

**Comment by AmirSavand** [comment]
No not blocked. Just want this in production.

**Comment by Alir3z4** [comment]
Alright let me try to reproduce the error and fix it.
This can go crazy when going on production

**Comment by AmirSavand** [comment]
What's the status @ArsalanSavand @Alir3z4

**Comment by Alir3z4** [comment]
I've got the solution,
It's because the solution we use in the backend we use to allow iframe is not compatible with all the browsers.

I'll make a patch for it in the backend then this can be merged.

**Comment by Alir3z4** [comment]
This is now fixed nicely.
I'm going to merge this now.

File: src/style/css/main.css [hunk 5]
```diff
@@ -1929,3 +1929,22 @@ img.cache {
     left: 250px !important;
     width: calc(100% - 250px) !important;
 }
+
+
+/* iframe */
+
+iframe.post-preview {
```
Comment (Alir3z4) [inline src/style/css/main.css]: Please move this to editor.css if this is not global.

**Comment by Alir3z4** [comment]
@ArsalanSavand 
I just made a tiny code change request.
Once that is being taken care of, please merge.

**Comment by ArsalanSavand** [comment]
@Alir3z4 Alright.

**Comment by ArsalanSavand** [comment]
It's good for merge.

**Comment by Alir3z4** [comment]
Great work!

**Pull request merged**